bonito/openvino/loader.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+from openvino.inference_engine import IECore, IENetwork
+
+import ngraph as ng
+from ngraph.impl.op import Parameter
+from ngraph.impl import Function, Shape, Type
+
+import torch
+from torch.autograd import Variable
+
+
+nodes = {}
+out = None
+
+def forward_hook(self, inputs, output):
+    global out
+    layer_type = self.__class__.__name__
+
+    params = [value.numpy() for value in self.state_dict().values()]
+
+    inp = nodes[inputs[0].data_ptr()]
+    if layer_type == 'Conv1d':
+        weights = np.expand_dims(params[0], axis=2)
+        if self.groups == 1:
+            out = ng.convolution(inp, weights,
+                                 [1, self.stride[0]],
+                                 [0, self.padding[0]],
+                                 [0, self.padding[0]],
+                                 [1, self.dilation[0]])
+
+        else:
+            weights = weights.reshape(self.groups, weights.shape[0] // self.groups, weights.shape[1], weights.shape[2], weights.shape[3])
+            out = ng.group_convolution(inp, weights,
+                                       [1, self.stride[0]],
+                                       [0, self.padding[0]],
+                                       [0, self.padding[0]],
+                                       [1, self.dilation[0]])
+        if len(params) > 1:
+            assert(len(params) == 2)
+            bias = params[1].reshape(1, params[1].shape[0], 1, 1)
+            out = ng.add(out, bias)
+
+    elif layer_type == 'BatchNorm1d':
+        out = ng.batch_norm_inference(inp, params[0], params[1], params[2], params[3], self.eps)
+    elif layer_type == 'Swish':
+        out = ng.swish(inp)
+    elif layer_type == 'Sum':
+        out = ng.add(inp, nodes[inputs[1].data_ptr()])
+    elif layer_type == 'Dropout':
+        return
+    else:
+        raise Exception('Unknown layer type: ' + layer_type)
+
+    nodes[output.data_ptr()] = out
+
+
+def sanity_check(net, inp, ref):
+    ie = IECore()
+    exec_net = ie.load_network(net, 'CPU')
+    ie_out = exec_net.infer({'input': inp.numpy()})
+    ie_out = next(iter(ie_out.values()))
+
+    ref = ref.numpy().reshape(ie_out.shape)
+    diff = np.max(np.abs(ie_out - ref))
+    print('PyTorch / OpenVINO diff:', diff)
+    print('Reference values range: [{}, {}]'.format(np.min(ref), np.max(ref)))
+    if diff > 1.1e-4:
+        raise Exception('Sanity check failed with diff', diff)
+
+
+def torch2openvino(model):
+    with torch.no_grad():
+        model.eval()
+        for module in model.modules():
+            if len([m for m in module.modules()]) != 1:
+                continue
+            module.register_forward_hook(forward_hook)
+
+        inp = Variable(torch.randn([1, 1, 1000]))
+
+        param = Parameter(Type.f32, Shape([1, 1, 1, 1000]))
+        nodes[inp.data_ptr()] = param
+        ref = model(inp)
+
+    out_node = ng.transpose(out, [0, 3, 2, 1])
+    out_node = ng.log(ng.softmax(out_node, axis=3))
```

Comment:
Breaking parity between MO and nG solution

Reply:
Do you mean that these two ops are out of general loop? Yes, it's a kind of issue for now because I could not catch functional ops by `register_forward_hook` (working on it)

Reply:
Well, seems like we cannot dynamically detect functional ops from PyTorch at this moment. That's why need to define these two operations manually.